### DISCUSSION TITLE
Add Location Information URL property to Wikidata items used by Name Suggestion Index

### DISCUSSION THREAD

**Discussion by davidhicks** [alltheplaces/alltheplaces#7245] [Ideas]
Run query:

https://query.wikidata.org/#SELECT%20%3Fbrand%20%3FbrandLabel%20%3Flocator_url%20WHERE%20%7B%0A%20%20SERVICE%20wikibase%3Alabel%20%7B%20bd%3AserviceParam%20wikibase%3Alanguage%20%22%5BAUTO_LANGUAGE%5D%2Cen%22.%20%7D%0A%20%20%3Fbrand%20wdt%3AP8253%20%3Fnsi_id.%0A%20%20OPTIONAL%20%7B%3Fbrand%20wdt%3AP12454%20%3Flocator_url%7D%0A%7D%0AGROUP%20BY%20%3Fbrand%20%3FbrandLabel%20%3Flocator_url

For brands with no [location information URL](https://www.wikidata.org/wiki/Property:P12454), check whether such a URL exists and then update the relevant Wikidata item to use this P12454 property to map brands/operators to location information pages/storefinder landing pages.

These URLs saved against brands/operators in Wikidata can be used in the future to automate the creation of spiders, and/or automate detection of changes to storefinder methods of a brand, etc.

**Comment by CloCkWeRX** [comment]
https://github.com/alltheplaces/alltheplaces/discussions/7155 - added all of this category into wikidata

**Comment by CloCkWeRX** [comment]
https://github.com/alltheplaces/alltheplaces/discussions/7060 mostly added

**Comment by CloCkWeRX** [comment]
```
SELECT ?brand ?brandLabel ?locator_url WHERE {
  SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],en". }
  ?brand wdt:P8253 ?nsi_id.
  ?brand wdt:P31 wd:Q507619.
  OPTIONAL {?brand wdt:P12454 ?locator_url}
}
GROUP BY ?brand ?brandLabel ?locator_url
```
Constrains this to only 'retail chain', which helps cut down things a bit - ~1000 entries.

**Reply by davidhicks** [reply]
Hopefully over time the Wikidata items for brands/retail chains will improve. Many are just instances of "business" or something equally unhelpful. Filtering on NSI ID (removing duplicates) at least is a strong indicator that the item is a brand or operator.

**Comment by CloCkWeRX** [comment]
470 populated, but I feel like lots of that is me.

**Comment by CloCkWeRX** [comment]
Up to 843.

The new tool for auto detecting links has a high success rate for finding these URLs; it would be worth setting that up as a wikidata bot to suggest urls.
I need a way to one-click or programatically add these claims; as manually is not very efficient (verify; edit, add statement, search, paste; save)
